codegen/generate_test.go

```diff
@@ -0,0 +1,92 @@
+package codegen
+
+import (
+	"testing"
+)
+
+// testPaths is a sampling of real paths that
+// originate in the OpenAPI doc, pulled via:
+// $ cat testdata/openapi.json | jq '.paths' | jq 'keys[]'
+var testPaths = []string{
+	"/database/roles",
+	"/database/roles/{name}",
+	"/auth/userpass/users/{username}/password",
+	"/auth/userpass/users/{username}/policies",
+	"/transit/export/{type}/{name}/{version}",
+}
+
+func TestCodeFilePath(t *testing.T) {
+	expected := []struct {
```

Comment:
This table test feels very flaky. 1) It's relying upon the ordering of the `testPaths` which are defined outside of this test. 2) You have to keep the number and order of test cases in line with the `testPaths` slice.

Instead, I recommend embedding the input path into the test case struct:
```go
[]struct {
  input string
  expectedDataSourcePath string
  expectedResourcePath string
}{
  {
    input: "/database/roles",
    expectedSourcePath: "/generated/datasources/database/roles.go",
    expectedResourcePath: "/generated/resources/database/roles.go",
  },
  ...
}
```
This helps ensure that adding/removing/changing test cases is self-contained and not reliant on external values that could be manipulated by other tests. It also makes the tests easier to understand because all of the values are in the test case struct definition and not in multiple locations.

Reply:
Sure, I could do that.

Reply:
This is done in https://github.com/terraform-providers/terraform-provider-vault/pull/738/commits/79a3b6c1e0ecf4dce6ccc947657b595c8b116965